pyk/src/pyk/proof/reachability.py

```diff
@@ -674,6 +674,7 @@ class APRProver(Prover[APRProof, APRProofStep, APRProofResult]):
     direct_subproof_rules: bool
     assume_defined: bool
     kcfg_explore: KCFGExplore
+    next_steps: dict[NodeIdLike, KCFGExtendResult]
```

Comment:
When we get two results back from `extend_cterm`, it is not clear that we can just apply them both, because, for example, applying the first result produces a bounded or a terminal node, and the execution should stop there. This is tested by the test suite and a number of tests that are supposed to pass will fail if we apply both.

My idea is to store this next step information in the prover and just re-use it when we explore from the node produced by the first step instead of running `extend_cterm` again. For this I've created the `next_steps` dictionary holding these next steps and the associated logic.